### PR TITLE
Remove support for 3.7 in development, update development dependencies

### PR DESCRIPTION
**Pull request by nllong**
Versions of flake8 and new BuildingsPy now require python 3.8

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,33 +29,33 @@ packages = [
...
-python = ">=3.7.1, <3.11"
-BuildingsPy = "3.0.0"
+python = ">=3.8, <3.11"
+BuildingsPy = "4.0.0"
 click = "8.1.3"
 geojson = "2.5.0"
 jinja2 = "3.1.2"
 jsonpath-ng = "1.5.3"
-jsonschema = "4.6.0"
+jsonschema = "~4.17"
```
Comment (vtnate) [inline pyproject.toml:38]: ❤️  tilde versioning

File: pyproject.toml [hunk 2]
```diff
@@ -29,33 +29,33 @@ packages = [
 
 [tool.poetry.dependencies]
 # Urbanopt SDK requires Python 3.10 as of UOv0.9.0
-python = ">=3.7.1, <3.11"
-BuildingsPy = "3.0.0"
+python = ">=3.8, <3.11"
```
Comment (vtnate) [inline pyproject.toml:32]: If we don't have to support 3.7, I think we can then support 3.11.

**Review by vtnate — APPROVED** [review]
Out with the old, in with the new!

File: pyproject.toml (hunk 1 above)
Comment (nllong) [reply pyproject.toml:38]: yeah, it's my favorite. So much better than `>`

**Pull request merged**